src/Business Foundation/App/NoSeries/src/Setup/NoSeriesSetupImpl.Codeunit.al

```diff
@@ -116,8 +116,10 @@ codeunit 305 "No. Series - Setup Impl."
             NoSeriesLine.SetRange("Series Code", NoSeriesRec.Code);
         end;
 
-        if not NoSeriesLine.FindFirst() then
+        if not NoSeriesLine.FindFirst() then begin
             NoSeriesLine.Init();
+            NoSeriesLine."Series Code" := NoSeriesRec.Code;
```

Comment:
Can we move the modification out of the procedure? `SetNoSeriesCurrentLineFilters`

Reply:
Then we should move the init out.

Reply:
Also there is no modify. We just make sure to init correctly, by setting Pk when init.